pkg/apis/admission/types.go

```diff
@@ -19,72 +19,73 @@ package admission
-	Name string
-	// Namespace is the namespace associated with the request (if any).
-	Namespace string
-	// Object is the object from the incoming request prior to default values being applied
-	Object runtime.Object
-	// OldObject is the existing object. Only populated for UPDATE requests.
-	OldObject runtime.Object
-	// Operation is the operation being performed
-	Operation Operation
 	// Resource is the name of the resource being requested.  This is not the kind.  For example: pods
 	Resource metav1.GroupVersionResource
 	// SubResource is the name of the subresource being requested.  This is a different resource, scoped to the parent
 	// resource, but it may have a different kind. For instance, /pods has the resource "pods" and the kind "Pod", while
 	// /pods/foo/status has the resource "pods", the sub resource "status", and the kind "Pod" (because status operates on
 	// pods). The binding resource for a pod though may be /pods/foo/binding, which has resource "pods", subresource
 	// "binding", and kind "Binding".
+	// +optional
 	SubResource string
+	// Name is the name of the object as presented in the request.  On a CREATE operation, the client may omit name and
+	// rely on the server to generate the name.  If that is the case, this method will return the empty string.
+	// +optional
+	Name string
+	// Namespace is the namespace associated with the request (if any).
+	// +optional
+	Namespace string
+	// Operation is the operation being performed
+	Operation Operation
 	// UserInfo is information about the requesting user
 	UserInfo authentication.UserInfo
-}
-
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-
-// AdmissionResponse describes an admission response.
-type AdmissionResponse struct {
-	metav1.TypeMeta
-
-	// Status is filled in by the webhook and indicates whether the admission
-	// request should be permitted.
+	// Object is the object from the incoming request prior to default values being applied
 	// +optional
-	Status AdmissionReviewStatus
-
-	// The json-patch body. Must be in compliance with RFC 6902.
+	Object runtime.Object
+	// OldObject is the existing object. Only populated for UPDATE requests.
 	// +optional
-	PatchBody []byte
+	OldObject runtime.Object
 }
 
-// AdmissionReviewStatus describes the status of the admission request.
-type AdmissionReviewStatus struct {
+// AdmissionResponse describes an admission response.
+type AdmissionResponse struct {
 	// Allowed indicates whether or not the admission request was permitted.
 	Allowed bool
 	// Result contains extra details into why an admission request was denied.
 	// This field IS NOT consulted in any way if "Allowed" is "true".
 	// +optional
 	Result *metav1.Status
+	// Patch contains the actual patch. Currently we only support a response in the form of JSONPatch, RFC 6902.
+	// +optional
+	Patch []byte
+	// PatchType indicates the form the Patch will take. Currently we only support "JSONPatch".
+	// +optional
+	PatchType string
```

Comment:
We often make special types for this, and define constants.

```go
type PatchType string

const (
  PatchTypeJSONPatch PatchType = "JSONPatch"
)
```

Reply:
Done